.github/workflows/publish.yml

```diff
@@ -0,0 +1,37 @@
+name: PyPI Automated publish
+
+on:
+  release:
+    types: [created]
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  deploy:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup python and poetry
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+        uses: Gr1N/setup-poetry@v8
+        run: poetry --version
```

Comment:
There seems to be a syntax issue with this line:

```
Invalid workflow file: .github/workflows/publish.yml#L21
The workflow is not valid. .github/workflows/publish.yml (Line: 21, Col: 9): 'uses' is already defined .github/workflows/publish.yml (Line: 22, Col: 9): Unexpected value 'run'
```

Reply:
Weird..That code was actually from their documentation. Should be fixed now though.

Reply:
Still getting an error: https://github.com/grapheneX/grapheneX/actions/runs/6966810649

```
Error: .github#L1
a step cannot have both the `uses` and `run` keys
```

Reply:
Fixed - CI finishes successfully now

Reply:
Thank you! One last comment :D